File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/axisAPIs/AxisRangingAutoRange.java
package com.scichart.docsandbox.examples.java.axisAPIs;

import androidx.annotation.NonNull;

import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.axes.AutoRange;
import com.scichart.charting.visuals.axes.NumericAxis;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class AxisRangingAutoRange extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) { }

    void setAutoRangeOnce() {
        // <SetAutoRangeOnce>
        final NumericAxis axis = new NumericAxis(getContext());
        axis.setAutoRange(AutoRange.Once);
        // </SetAutoRangeOnce>
    }

    void setAutoRangeAlways() {
        // <SetAutoRangeAlways>
        final NumericAxis axis = new NumericAxis(getContext());
        axis.setAutoRange(AutoRange.Always);
        // </SetAutoRangeAlways>
    }

    void setAutoRangeNever() {
        // <SetAutoRangeNever>
        final NumericAxis axis = new NumericAxis(getContext());
        axis.setAutoRange(AutoRange.Never);
        // </SetAutoRangeNever>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/axisAPIs/AxisRangingAutoRange.kt
package com.scichart.docsandbox.examples.kotlin.axisAPIs

import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.axes.AutoRange
import com.scichart.charting.visuals.axes.NumericAxis
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition()
class AxisRangingAutoRange : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {}

    fun setAutoRangeOnce() {
        // <SetAutoRangeOnce>
        val axis = NumericAxis(context)
        axis.autoRange = AutoRange.Once
        // </SetAutoRangeOnce>
    }

    fun setAutoRangeAlways() {
        // <SetAutoRangeAlways>
        val axis = NumericAxis(context)
        axis.autoRange = AutoRange.Always
        // </SetAutoRangeAlways>
    }

    fun setAutoRangeNever() {
        // <SetAutoRangeNever>
        val axis = NumericAxis(context)
        axis.autoRange = AutoRange.Never
        // </SetAutoRangeNever>
    }
}
